geoportailv3/static/js/layerinfo/layerinfoservice.js

```diff
@@ -14,32 +14,112 @@ goog.require('ngeo.CreatePopup');
 app.ShowLayerinfo;
 
 
+
 /**
  * @param {angular.$http} $http Angular $http service
  * @param {angular.$sce} $sce Angular $sce service
+ * @param {angular.Scope} $rootScope The rootScope provider.
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog.
  * @param {ngeo.CreatePopup} ngeoCreatePopup Ngeo popup factory service
- * @return {app.ShowLayerinfo} The show layer info function.
+ * @constructor
  * @ngInject
  */
-app.showLayerinfoFactory = function($http, $sce, ngeoCreatePopup) {
+app.showLayerinfoFactory = function($http, $sce, $rootScope,
+    gettextCatalog, ngeoCreatePopup) {
 
   /**
    * @type {ngeo.Popup}
    */
   var popup = ngeoCreatePopup();
+  /**
+   * @type {Object.<string, !angular.$q.Promise>}
+   * @private
+   */
+  var promises_ = {};
+
+  /**
+   * @type {ol.layer.Layer}
+   */
+  var currentLayer = null;
+
+  $rootScope.$on('gettextLanguageChanged', function() {
+    if (!goog.isNull(currentLayer) && popup.getOpen()) {
+      showLayerInfo(currentLayer);
+    }
+  });
+
+  return showLayerInfo;
+
+  /**
+   * @param {ol.layer.Layer} layer The layer
+   * @this {Object}
+   */
+  function showLayerInfo(layer) {
+    currentLayer = layer;
+    var title = /** @type {string} */ (layer.get('label'));
+    var localMetadata = /** @type {Object.<string>} */
+        (layer.get('metadata'));
```

Comment:
The type syntax you use for `title` and `localMetadata` is a type cast, which is totally correct in that case, because `layer.get` is a generic function whose [return type is `*`](https://github.com/openlayers/ol3/blob/master/src/ol/object.js#L351).